SU2_CFD/src/solvers/CSpeciesSolver.cpp

```diff
@@ -0,0 +1,721 @@
+
+      /*--- Compute the residual using an upwind scheme ---*/
+
+      auto residual = conv_numerics->ComputeResidual(config);
+      LinSysRes.AddBlock(iPoint, residual);
+
+      /*--- Jacobian contribution for implicit integration ---*/
+      const bool implicit = (config->GetKind_TimeIntScheme() == EULER_IMPLICIT);
+      if (implicit) Jacobian.AddBlock2Diag(iPoint, residual.jacobian_i);
+
+      //      /*--- Viscous contribution, commented out because serious convergence problems ---*/
+      //
+      //      su2double Coord_Reflected[MAXNDIM];
+      //      GeometryToolbox::PointPointReflect(nDim, geometry->nodes->GetCoord(Point_Normal),
+      //                                               geometry->nodes->GetCoord(iPoint), Coord_Reflected);
+      //      visc_numerics->SetCoord(geometry->nodes->GetCoord(iPoint), Coord_Reflected);
+      //      visc_numerics->SetNormal(Normal);
+      //
+      //      /*--- Conservative variables w/o reconstruction ---*/
+      //
+      //      visc_numerics->SetPrimitive(V_domain, V_inlet);
+      //
+      //      /*--- Turbulent variables w/o reconstruction, and its gradients ---*/
+      //
+      //     visc_numerics->SetScalarVar(Solution_i, Solution_j);
+      //     visc_numerics->SetScalarVarGradient(node[iPoint]->GetGradient(), node[iPoint]->GetGradient());
+      //
+      //      /*--- Menter's first blending function ---*/
+      //
+      //      visc_numerics->SetF1blending(node[iPoint]->GetF1blending(), node[iPoint]->GetF1blending());
+      //
+      //      /*--- Compute residual, and Jacobians ---*/
+      //
+      //      auto residual = visc_numerics->ComputeResidual(config);
+      //
+      //      /*--- Subtract residual, and update Jacobians ---*/
+      //
+      //      LinSysRes.SubtractBlock(iPoint, residual);
+      //      Jacobian.SubtractBlock2Diag(iPoint, residual.jacobian_i);
+    }
+  }
+  END_SU2_OMP_FOR
+}
+
+void CSpeciesSolver::SetInletAtVertex(const su2double *val_inlet,
+                                      unsigned short iMarker,
+                                      unsigned long iVertex) {
+
+  for (unsigned short iVar = 0; iVar < nVar; iVar++)
+    Inlet_SpeciesVars[iMarker][iVertex][iVar] = val_inlet[Inlet_Position+iVar];
+
+}
+
+su2double CSpeciesSolver::GetInletAtVertex(su2double *val_inlet,
+                                           unsigned long val_inlet_point,
+                                           unsigned short val_kind_marker,
+                                           string val_marker,
+                                           const CGeometry *geometry,
+                                           const CConfig *config) const {
+  /*--- Local variables ---*/
+
+  unsigned short iMarker;
+  unsigned long iPoint, iVertex;
+  su2double Area = 0.0;
+  su2double Normal[3] = {0.0,0.0,0.0};
+
+  /*--- Alias positions within inlet file for readability ---*/
+
+  if (val_kind_marker == INLET_FLOW) {
```

Comment:
```suggestion
  if (val_kind_marker != INLET_FLOW) return Area;
```

Reply:
Good idea, but we have this implementation currently 4 times (meanFlow, SA, SST and now Species). These inlet-specific functions are maybe a good candidate to be generalized into CScalarSolver at some point. And to keep everything as comparable as possible I would prefer not to change this alone